qcodes/instrument/parameter.py

```diff
@@ -533,28 +533,30 @@ def _from_raw_value_to_value(self, raw_value: ParamRawDataType
         # apply offset first (native scale)
         if self.offset is not None:
             # offset values
-            if isinstance(self.offset, collections.abc.Iterable):
-                # offset contains multiple elements, one for each value
-                value = tuple(val - offset for val, offset
-                              in zip(value, self.offset))
-            elif isinstance(value, collections.abc.Iterable):
-                # Use single offset for all values
-                value = tuple(val - self.offset for val in value)
-            else:
-                value -= self.offset
+            try:
+                value = value - self.offset
+            except TypeError:
+                if isinstance(self.offset, collections.abc.Iterable):
+                    # offset contains multiple elements, one for each value
+                    value = tuple(val - offset for val, offset
+                                  in zip(value, self.offset))
+                elif isinstance(value, collections.abc.Iterable):
+                    # Use single offset for all values
+                    value = tuple(val - self.offset for val in value)
```

Comment:
what happens on an implicit else here? shall we reraise the TypeError at least
```suggestion
                    value = tuple(val - self.offset for val in value)
                raise
```

Reply:
yes, i'll add it

Reply:
updated